src/main.rs

```diff
@@ -203,6 +203,12 @@ fn get_files_in_dir(paths: &mut Vec<PathBuf>, path: PathBuf) {
     paths.extend(temp_paths);
 }
 
+#[cfg(not(feature = "git"))]
+fn git_repos(options: &Options, args: &[&OsStr]) -> bool {
+    return false;
+}
+
+#[cfg(feature = "git")]
 fn git_repos(options: &Options, args: &[&OsStr]) -> bool {
     let option_enabled = match options.view.mode {
```

Comment:
```suggestion
fn git_repos(options: &Options, args: &[&OsStr]) -> bool {
    if cfg!(not(feature = "git")) {
       return false;
    }
    let option_enabled = match options.view.mode {
```

Reply:
@MartinFillon your suggestion does not fix the error, I still get
```
error[E0433]: failed to resolve: use of undeclared crate or module `git2`
   --> src/main.rs:234:18
    |
234 |             .map(git2::Repository::open)
    |                  ^^^^ use of undeclared crate or module `git2`
    |
help: there is a crate or module with a similar name
    |
234 |             .map(git::Repository::open)
    |                  ~~~

For more information about this error, try `rustc --explain E0433`.
error: could not compile `eza` (bin "eza") due to previous error
```

using the patch as is just gives these warnings:
```
   Compiling eza v0.16.2 (/mnt/data/Users/leohd/source/repos/eza)
warning: unused variable: `options`
   --> src/main.rs:207:14
    |
207 | fn git_repos(options: &Options, args: &[&OsStr]) -> bool {
    |              ^^^^^^^ help: if this is intentional, prefix it with an underscore: `_options`
    |
note: the lint level is defined here
   --> src/main.rs:7:9
    |
7   | #![warn(unused)]
    |         ^^^^^^
    = note: `#[warn(unused_variables)]` implied by `#[warn(unused)]`

warning: unused variable: `args`
   --> src/main.rs:207:33
    |
207 | fn git_repos(options: &Options, args: &[&OsStr]) -> bool {
    |                                 ^^^^ help: if this is intentional, prefix it with an underscore: `_args`

warning: function `get_files_in_dir` is never used
   --> src/main.rs:194:4
    |
194 | fn get_files_in_dir(paths: &mut Vec<PathBuf>, path: PathBuf) {
    |    ^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` implied by `#[warn(unused)]`

warning: `eza` (bin "eza") generated 3 warnings (run `cargo fix --bin "eza"` to apply 2 suggestions)
    Finished release [optimized] target(s) in 22.37s
```